// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package chat;

// The greeting service definition.
service Chatter {
  // Sends a greeting
  rpc SendChat (ChatMessage) returns (Ack) {}
}

// The request message containing the user's name.
message ChatMessage {
  string name = 1;
  string msg = 2;
}

message Ack {
  int32 error_code = 1;
}
